src/isar/state_machine/state_machine.py

```diff
@@ -64,23 +65,110 @@ def __init__(
+            self.idle_state,
+            self.initiate_step_state,
+            self.monitor_state,
+            self.stop_state,
+            self.paused_state,
         ]
+
         self.machine = Machine(
             self,
             states=self.states,
             initial="off",
             queued=True,
         )
+
+        self.machine.add_transitions(
+            [
+                {
+                    "trigger": "step_initiated",
+                    "source": self.initiate_step_state,
+                    "dest": self.monitor_state,
+                    "before": self._step_initiated,
+                },
+                {
+                    "trigger": "pause",
+                    "source": [self.initiate_step_state, self.monitor_state],
+                    "dest": self.stop_state,
+                    "before": self._pause,
+                },
+                {
+                    "trigger": "stop",
+                    "source": [self.initiate_step_state, self.monitor_state],
+                    "dest": self.stop_state,
+                    "before": self._stop,
+                },
+                {
+                    "trigger": "finalize",
+                    "source": [
+                        self.initiate_step_state,
+                        self.stop_state,
+                        self.paused_state,
+                    ],
+                    "dest": self.idle_state,
+                    "before": self._finalize,
+                },
+                {
+                    "trigger": "mission_started",
+                    "source": self.idle_state,
+                    "dest": self.initiate_step_state,
+                    "before": self._mission_started,
+                },
+                {
+                    "trigger": "unpause",
+                    "source": self.paused_state,
+                    "dest": self.initiate_step_state,
+                    "before": self._unpause,
+                },
+                {
+                    "trigger": "step_finished",
+                    "source": self.monitor_state,
+                    "dest": self.initiate_step_state,
+                    "before": self._step_finished,
+                },
+                {
+                    "trigger": "paused_successfully",
+                    "source": self.monitor_state,
+                    "dest": self.initiate_step_state,
+                    "before": self._paused_successfully,
+                },
+                {
+                    "trigger": "step_infeasible",
+                    "source": self.initiate_step_state,
+                    "dest": self.initiate_step_state,
+                    "before": self._step_infeasible,
+                },
+                {
+                    "trigger": "step_failed",
+                    "source": self.initiate_step_state,
+                    "dest": self.idle_state,
+                    "before": self._step_failed,
+                },
```

Comment:
What is actually happening in this transition? It's not `step_failed`?

I assume it's meant that we failed to schedule anything on the robot the ten times and thus cancel the mission?

Reply:
Guess `initiate_step_failed` would be a more descriptive name